### DISCUSSION TITLE
Potential memory leak in v-autocomplete component

### DISCUSSION THREAD

**Discussion by v1talii-dev** [vuetifyjs/vuetify#15788] [General]
First I go to the page [https://v15.vuetifyjs.com/en/components/autocompletes](https://v15.vuetifyjs.com/en/components/autocompletes) then I go to the main page.

I notice separated elements in the developer tools. Can this be considered a memory leak?

in my project I use version 1.5.22, I have the same behavior.

![2022-09-16_12-53-14](https://user-images.githubusercontent.com/12966923/190617201-e2e320ac-53ad-4ff3-a0e5-1b3c9f2bcd9b.png)

**Comment by KaelWD** [comment]
Are they still present if you trigger garbage collection?

1.5 is no longer supported anyway, if this is a real bug it would only be fixed in v2

**Reply by v1talii-dev** [reply]
Yes, I started the garbage collector. The screenshot shows from the detached nodes after starting the garbage collector.

**Reply by v1talii-dev** [reply]
I tested version 2.6.10 and see detached elements...
![image](https://user-images.githubusercontent.com/12966923/190627082-7e8a731f-2f86-4eb1-94aa-299fb479c0ce.png)

**Reply by v1talii-dev** [reply]
I made memory snapshots using the following algorithm. Went to the page [autocompletes](https://vuetifyjs.com/en/components/autocompletes). Then to the main page, took a snapshot. And he repeated it three times. In the screenshot, you can see that memory increases between snapshots. Yes, it's probably a project problem with documentation, but it's more likely that it's a problem directly with vuetify.

![image](https://user-images.githubusercontent.com/12966923/190632760-c541a010-caf8-4b84-94dd-403c73f70a41.png)

![image](https://user-images.githubusercontent.com/12966923/190633368-5a1c8759-06d6-4b82-a12f-e026566c47df.png)

**Reply by KaelWD** [reply]
- If you have an adblocker enabled some are retained by the stack traces, clearing the console allows these to be garbage collected
- Some are retained by the search bar not cleaning up after itself properly, fixed by 763af05a0
- Some are retained by the theme selector not cleaning up after itself properly, fixed by 99e726647
- Some are retained by the click-outside directive keeping a stale vnode reference in scope, fixed by 4d3359ad5

Heap size seems to be stable around 60MB now instead of climbing into the hundreds after a few navigations.

**Reply by v1talii-dev** [reply]
Thanks for the quick response and corrections. Tell me, please, is there a problem with detached elements?
P.s.: I did not use an ad blocker in the browser in which the tests were conducted. Of the plug-ins connected, there was only Vue.js devtools

**Reply by v1talii-dev** [reply]
it can't be related to these problems? https://github.com/vuejs/vue/pull/10085 and https://github.com/vuejs/vue/issues/10004?
